pkg/virt-api/webhooks/validating-webhook/admitters/migration-create-admitter_test.go

```diff
@@ -34,91 +33,65 @@ import (
 	"kubevirt.io/client-go/api"
 
 	v1 "kubevirt.io/api/core/v1"
-	"kubevirt.io/client-go/kubecli"
+	kubevirtfake "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/fake"
 
 	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
 )
 
 var _ = Describe("Validating MigrationCreate Admitter", func() {
-	// Mock VirtualMachineInstanceMigration
-	var ctrl *gomock.Controller
-	var virtClient *kubecli.MockKubevirtClient
+	var virtClient *kubevirtfake.Clientset
 	var migrationCreateAdmitter *MigrationCreateAdmitter
-	var migrationInterface *kubecli.MockVirtualMachineInstanceMigrationInterface
-	var mockVMIClient *kubecli.MockVirtualMachineInstanceInterface
 
 	BeforeEach(func() {
-		ctrl = gomock.NewController(GinkgoT())
-		migrationInterface = kubecli.NewMockVirtualMachineInstanceMigrationInterface(ctrl)
-		mockVMIClient = kubecli.NewMockVirtualMachineInstanceInterface(ctrl)
-		virtClient = kubecli.NewMockKubevirtClient(ctrl)
-		virtClient.EXPECT().VirtualMachineInstanceMigration("default").Return(migrationInterface).AnyTimes()
-		virtClient.EXPECT().VirtualMachineInstance(gomock.Any()).Return(mockVMIClient).AnyTimes()
+		virtClient = kubevirtfake.NewSimpleClientset()
+		Expect(virtClient.Fake.Resources).To(BeEmpty())
 		migrationCreateAdmitter = NewMigrationCreateAdmitter(virtClient)
 	})
 
 	It("should reject Migration spec on create when another VMI migration is in-flight", func() {
 		vmi := api.NewMinimalVMI("testmigratevmi2")
+		virtClient.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Create(context.Background(), vmi, metav1.CreateOptions{})
+
 		inFlightMigration := v1.VirtualMachineInstanceMigration{
 			ObjectMeta: metav1.ObjectMeta{
 				Namespace: vmi.Namespace,
+				Labels:    map[string]string{v1.MigrationSelectorLabel: vmi.Name},
```

Comment:
Could you please add reasoning for this addition? (IIRC it is the result of the migration mutator)

Reply:
Fakeclient doesn't pass it through mutation so it never hits `addMigrationSelectorLabel()`, I could add a prepend reactor but that would be an overkill IMHO so I'll be keeping it like so otherwise the test can't run properly.

Reply:
I meant adding an explanation to the commit message why is this change needed.